### PR TITLE
New Mesh: EC15to60E2r4 

### PR DESCRIPTION
**Pull request by xylar**
Long Name: EC15to60kmL60E3SMv2r04

MPAS Eddy Closure mesh for E3SM version2  with enhanced resolution around the equator (30 km), South pole (35 km), Greenland (15 km ), 60-km resolution at mid latitudes, and 60 vertical levels.  Bathymetry is from GEBCO 2019, combined with BedMachineAntarctica around Antarctica.  The default initial condition is from EN4 1900, but PHC is also an option.

The motivation for this new mesh is enhanced resolution around Greenland, similar to the EC15to60E2r1 mesh (https://github.com/MPAS-Dev/MPAS-Model/pull/522), that should hopefully increase the strength of the AMOC, which was weak in E3SM v1 and is even weaker in E3SM v2.

### CODE REVIEW

**Comment by xylar** [comment]
I think I should rebase this to take out all the EC30to60 changes.

**Comment by xylar** [comment]
## First draft

cellWidthGlobal

**Comment by vanroekel** [comment]
Thanks @xylar this is looking great.  Thanks for pulling this together so quickly.  @maltrud @proteanplanet @milenaveneziani @lconlon could you take a look as well?

**Comment by xylar** [comment]
I will post some plots of the culled mesh as soon as I have it.

**Comment by proteanplanet** [comment]
> I will post some plots of the culled mesh as soon as I have it.

Please point me to the mesh file when you have it.

**Comment by xylar** [comment]
> Please point me to the mesh file when you have it.

I'll do that.  Grizzly is down (or at least I can't log in) so I'll be making the mesh on Anvil.  Hopefully, the fact that there's no space won't kill this work...

**Comment by xylar** [comment]
## Results on Anvil

The E3SM files with metadata are not quite ready yet.  The file system seems unusually slow, perhaps because the disk is quite full?

The culled mesh is here:
/home/ac.xylar/acme/mpas/spinup_EC15to60E2r4/ocean/global_ocean/EC15to60/build_mesh/culled_mesh/culled_mesh.nc

The initial state (before any spin-up) is here:
/home/ac.xylar/acme/mpas/spinup_EC15to60E2r4/ocean/global_ocean/EC15to60/spin_up_EN4_1900/initial_state/initial_state.nc

The restart file after a 1-hour spin-up is here:
/home/ac.xylar/acme/mpas/spinup_EC15to60E2r4/ocean/global_ocean/EC15to60/spin_up_EN4_1900/simulation/restarts/restart.0001-01-01_01.00.00.nc

I'll update here once the ocean/sea ice data in the format needed for E3SM and with metadata for the mesh are ready.

**Comment by proteanplanet** [comment]
@xylar Please can you change permissions on the initial_state.nc file above?

**Comment by xylar** [comment]
@proteanplanet, the permissions look correct to me:
```
$ ls -lah -lah initial_state.nc 
-rw-r-----+ 1 ac.xylar cels 2.5G Nov  3 20:59 initial_state.nc
```
I'm guessing the problem was that I use a link from my home directory to get to `/lcrc/group/acme/ac.xylar/` and you won't have access to that.  I changed the paths above to the absolute path without that link.  Could you see if you have access to that path?

**Comment by proteanplanet** [comment]
@xylar I can't work this out.  I can't access /lcrc/group/acme/ac.xylar, but can access other of my colleague's home directories at the same place.  We are in the same groups.  If you can copy the initial_state.nc to /lcrc/group/acme/ac.afroberts, please do.

**Comment by xylar** [comment]
@proteanplanet, I changed the group for `/lcrc/group/acme/ac.xylar` from `climate` to `cels`, so maybe that will help.

Just in case, I also copied `initial_state.nc` to: `/lcrc/group/acme/ac.afroberts/EC15to60E2r4_initial_state.nc`.

**Comment by xylar** [comment]
It is also 6:30 am and I'm becoming a bit delirious so I don't think it's safe for me to work on this anymore.

**Comment by proteanplanet** [comment]
Looks good to me.  Glad you went to bed @xylar 

EC15to60E2r4_sector_mesh_7

EC15to60E2r4_sector_mesh_1

**Comment by vanroekel** [comment]
Thanks @proteanplanet for the images.  The mesh looks great @xylar, thanks for working on this so quickly.  All (@milenaveneziani @lconlon @mark-petersen @sbrus89 @maltrud) can you please enter reviews as soon as possible?  I'd like to start testing this mesh asap.

**Review by vanroekel — APPROVED** [review]
approved by visual inspection and images from @xylar and @proteanplanet

**Review by milenaveneziani — APPROVED** [review]

**Review by lconlon — APPROVED** [review]
Looks good. Thanks!!

**Review by sbrus89 — APPROVED** [review]
Looks great to me.

**Comment by xylar** [comment]
@mark-petersen, this is a PR that we want to put in before moving over to the COMPASS repo (#682).  It is very time critical.

**Comment by jonbob** [comment]
@xylar - please let me know when you feel like there's a mesh file I can start making mapping files from. And thanks for all your hard work!

**Comment by xylar** [comment]
@jonbob, the info you need should be in place.  Hopefully, what you need is in:
```
/lcrc/group/acme/ac.xylar/mpas/spinup_EC15to60E2r4/ocean/global_ocean/EC15to60/files_for_e3sm/files_for_e3sm/assembled_files_for_upload/
```
Please let me know if that isn't what you need.

**Comment by jonbob** [comment]
All I really need is the initial file -- I can make everything else from that. So I'll get started!

**Comment by xylar** [comment]
@jonbob, it would be great if the files that end up on the LCRC server have the metadata that gets added during the `files_for_e3sm` step form COMPASS.  Do you know if that metadata is preserved in the operations that you do?
This is the metadata that I mean:
```
		:MPAS_Mesh_Short_Name = "EC15to60E2r04" ;
		:MPAS_Mesh_Long_Name = "EC15to60kmL60E3SMv2r04" ;
		:MPAS_Mesh_Description = "MPAS Eddy Closure mesh for E3SM versio
n 2 with enhanced resolution around the equator (30 km), South pole (35 km), Gre
enland (15 km), 60-km resolution at mid latitudes, and 60 vertical levels.  Bath
ymetry is from GEBCO 2019, combined with BedMachineAntarctica around Antarctica.
 The initial condition is from EN4 1900.  The mesh is documented at https://gith
ub.com/MPAS-Dev/MPAS-Model/pull/740" ;
		:MPAS_Mesh_E3SM_Version = 2 ;
		:MPAS_Mesh_EC_Version = "04" ;
		:MPAS_Mesh_EC_Version_Author = "Xylar Asay-Davis" ;
		:MPAS_Mesh_EC_Version_Author_E-mail = "xylar@lanl.gov" ;
		:MPAS_Mesh_EC_Version_Creation_Date = "11/03/2020 21:34:39" ;
		:MPAS_Mesh_EC_Minimum_Resolution_km = 15. ;
		:MPAS_Mesh_EC_Maximum_Resolution_km = 60. ;
		:MPAS_Mesh_EC_Maximum_Depth_m = 5500. ;
		:MPAS_Mesh_EC_Number_of_Levels = 60. ;
		:MPAS_Mesh_Ice_Shelf_Cavities = "OFF" ;
		:MPAS_Mesh_Runoff_Description = "<<<Spreading function described
 here>>>" ;
		:MPAS_Mesh_COMPASS_Version = "0.1.11" ;
		:MPAS_Mesh_JIGSAW_Version = "0.9.12" ;
		:MPAS_Mesh_JIGSAW-Python_Version = "0.2.1" ;
		:MPAS_Mesh_MPAS-Tools_Version = "0.0.13" ;
		:MPAS_Mesh_NCO_Version = "4.9.4" ;
		:MPAS_Mesh_ESMF_Version = "8.0.1" ;
		:MPAS_Mesh_geometric_features_Version = "0.1.11" ;
		:MPAS_Mesh_Metis_Version = "5.1.0" ;
		:MPAS_Mesh_pyremap_Version = "0.0.7" ;
```

**Comment by jonbob** [comment]
@xylar - I'll check and modify my workflow if needed to get the metadata. For what it's worth, your version of the tool to create the SCRIP file still has issues with missing values in the grid_imask field?

**Comment by jonbob** [comment]
@xylar - that metadata is not included in the SCRIP files, at least not automatically. So it would be lost for all mapping files

**Review by proteanplanet — APPROVED** [review]
This is approved.  

Note: The resolution transition through the Archileago is barely noticable:

EC15to60E2r4_zoom_bathymetry_1

For interest only, with no meaningful impact on the applications of this particular mesh except for show and tell purposes, updates to geometric features since previous meshes now includes:

1) The Hawaiin island chain is reliably represented:

EC15to60E2r4_zoom_bathymetry_41

2) British Isles more accurately resolved:

EC15to60E2r4_zoom_bathymetry_17

3) And Italy reliably with a boot:

EC15to60E2r4_zoom_bathymetry_19

**Comment by xylar** [comment]
> I'll check and modify my workflow if needed to get the metadata. For what it's worth, your version of the tool to create the SCRIP file still has issues with missing values in the grid_imask field?

Oh, no!  I think that many simply be because we haven't had a release of MPAS-Tools since the bug fix but that's a good reminder that we need one!  Sorry about that!

>  that metadata is not included in the SCRIP files, at least not automatically. So it would be lost for all mapping files

That's a good point.  I believe I can add it but I hadn't been.  But I don't think it's critical that it be included in the mapping files.  What really matters is if it propagates to E3SM output files.  And that may depend more on MPAS-Model than on anything we do.  The main thing is to make sure it's in whatever initial condition files get put on the LCRC server.  I'm not sure if that's you, @jonbob or @vanroekel or someone else...

**Comment by jonbob** [comment]
@xylar - OK, I'll use my version of the SCRIP file, so no problem. And I won't worry about metadata in the mapping files, which means I get get started right away. And we'll pay attention to the ocean initial file that gets committed. Thanks again for everything

**Review by maltrud — APPROVED** [review]

File: testing_and_setup/compass/ocean/global_ocean/EC15to60/build_mesh/build_base_mesh.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+def main():
+    cellWidth, lon, lat = cellWidthVsLatLon()
+    build_spherical_mesh(cellWidth, lon, lat, out_filename='base_mesh.nc')
+
+
+if __name__ == '__main__':
+    main()
+
+
+numpy.nan*xarray.ones_like()
```
Comment (mark-petersen) [inline testing_and_setup/compass/ocean/global_ocean/EC15to60/build_mesh/build_base_mesh.py:84]: This looks like a paste typo. Should I delete it?

Comment (xylar) [reply testing_and_setup/compass/ocean/global_ocean/EC15to60/build_mesh/build_base_mesh.py:84]: Oops, I'll remove this.

Comment (xylar) [reply testing_and_setup/compass/ocean/global_ocean/EC15to60/build_mesh/build_base_mesh.py]: Fixed.

**Review by mark-petersen — APPROVED** [review]
I tested on grizzly. Thanks!

**Pull request merged**